File: src/autosave/AutosaveCooldownManager.java
package autosave;

import com.fs.starfarer.api.Global;
import lunalib.lunaSettings.LunaSettings;

public class AutosaveCooldownManager {

    private static long lastAutosaveTimeMs = System.currentTimeMillis();

    public static boolean canAutosave() {
        float minutes = LunaSettings.getFloat("autosave_extended", "autosave_min_interval");
        long minIntervalMs = (long)(minutes * 60_000);

        long now = System.currentTimeMillis();
        return now - lastAutosaveTimeMs >= minIntervalMs;
    }

    public static void markSaved() {
        lastAutosaveTimeMs = System.currentTimeMillis();
    }
}

File: src/autosave/AutosaveCooldownManager.kt
package autosave

import lunalib.lunaSettings.LunaSettings

object AutosaveCooldownManager {

    private var lastAutosaveTimeMs: Long = System.currentTimeMillis()

    fun canAutosave(): Boolean {
        val minutes = LunaSettings.getFloat("autosave_extended", "autosave_min_interval")
        val minIntervalMs = (minutes * 60_000).toLong()

        val now = System.currentTimeMillis()
        return now - lastAutosaveTimeMs >= minIntervalMs
    }

    fun markSaved() {
        lastAutosaveTimeMs = System.currentTimeMillis()
    }
}
